src/MemoryStrategy.ts

```diff
@@ -0,0 +1,46 @@
+import { Filter, SetCallback, DBRecord } from './collection';
+
+export class MemoryStrategy<T extends DBRecord> {
+  private data: any[];
+
+  constructor(data: any[]) {
+    this.data = data || [];
+  }
+
+  public async getAll(): Promise<T[]> {
+    return [...this.data];
+  }
+
+  public async getData(callback: Filter<T>): Promise<T[]> {
+    return this.data.filter(callback);
+  }
+
+  public async insert(newDocument: T): Promise<T> {
+    this.data = [...this.data, newDocument];
+    return newDocument;
+  }
+
+  public async update<K extends T>(
+    filter: Filter<K>,
+    modifier: SetCallback<T>,
+  ): Promise<boolean> {
+    let updated = false;
+    const newDocuments = this.data.filter(filter).map((document) => {
+      updated = true;
+      const documentId = document.id;
+      const newDocument = { ...modifier(document), id: documentId };
+      return newDocument;
+    });
+    this.data = newDocuments;
+    return updated;
+  }
+
+  public async delete(filter: Filter<T>): Promise<boolean> {
+    const newData = this.data.filter((document) =>
```

Comment:
how about just `this.data.filter(filter)`?

Reply:
It could be confusing, because we want to Delete filtered items, and your way leaves them and removes all the rest.

Reply:
oh, I see, thanks

Reply:
nit-picky: then maybe `!filter(document)`?